pinot-core/src/main/java/org/apache/pinot/core/common/datablock/DataBlockBuilder.java

```diff
@@ -151,7 +151,11 @@ public static RowDataBlock buildFromRows(List<Object[]> rows, DataSchema dataSch
             byteBuffer.putInt(((Boolean) value) ? 1 : 0);
             break;
           case TIMESTAMP:
-            byteBuffer.putLong(((Timestamp) value).getTime());
+            if (value instanceof Long) {
```

Comment:
Are we going to hit both? If so, can you add some java doc explaining when will we hit both? If we can hit both, then `BOOLEAN` will also break

Reply:
Certain non strong typed functions in v2 might return long value then in datablock build, we need to do the conversion.